### PR TITLE
feat(community-callout-paragraph): add icon support

### PR DESCRIPTION
**Pull request by elissa-matsushita**
Add optional icon before text

## Related issues

See #451 

## Description

- Add new 'icon' prop
- if this prop is supplied, it will display before the text

## Checklist before submitting pull request

- Commits follow our [Developer Guide](https://github.com/telus/tds-community/blob/chore/template-hints/.github/CONTRIBUTING.md)
- [x] New code is unit tested
- [x] make sure visual and accessibility tests pass
- [x] make sure code builds

### CODE REVIEW

File: packages/CalloutParagraph/CalloutParagraph.jsx [hunk 1]
```diff
@@ -59,17 +61,46 @@ const TextWrapper = styled.div`
...
+  }
+
   return (
-    <TextWrapper spacing={spacing} roundedCorners={roundedCorners} compact={compact}>
-      <Paragraph {...safeRest(rest)}>{children}</Paragraph>
+    <TextWrapper
+      spacing={spacing}
+      roundedCorners={roundedCorners}
+      compact={compact}
+      icon={icon}
```
Comment (Andrew-K-Lam) [inline packages/CalloutParagraph/CalloutParagraph.jsx:97]: Why is the icon prop being passed to TextWrapper.  It doesnt look like its being used

**Review by Andrew-K-Lam — CHANGES_REQUESTED** [review]

File: packages/CalloutParagraph/CalloutParagraph.jsx (hunk 1 above)
Comment (elissa-matsushita) [reply packages/CalloutParagraph/CalloutParagraph.jsx]: Good question! This has been updated :) Thanks for catching that!

File: packages/CalloutParagraph/CalloutParagraph.jsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { safeRest } from '@tds/util-helpers'
 
 import Paragraph from '@tds/core-paragraph'
 import { colorTelusPurple } from '@tds/core-colours'
+import * as DecorativeIcons from '@tds/core-decorative-icon'
```
Comment (jraff) [inline packages/CalloutParagraph/CalloutParagraph.jsx:9]: Can we pass the icon element directly via the `icon` prop?

See Link icons - https://tds.telus.com/components/index.html#!/Links/Link/5

File: packages/CalloutParagraph/CalloutParagraph.jsx [hunk 3]
```diff
@@ -59,17 +61,45 @@ const TextWrapper = styled.div`
       padding-left: ${props => paddingValue.mobile[props.spacing]};
       padding-right: ${props => paddingValue.mobile[props.spacing]};
     }
+    > i {
```
Comment (jraff) [inline packages/CalloutParagraph/CalloutParagraph.jsx:64]: We will soon be dropping support for font icons and this css will not be needed. We can keep it for now, just something to note.

File: packages/CalloutParagraph/CalloutParagraph.jsx [hunk 4]
```diff
@@ -59,17 +61,45 @@ const TextWrapper = styled.div`
...
 `
 
 /**
  * @version ./package.json
  * @visibleName CalloutParagraph (beta)
  */
-const CalloutParagraph = ({ children, spacing, roundedCorners, compact, ...rest }) => {
+const CalloutParagraph = ({ children, spacing, roundedCorners, compact, icon, ...rest }) => {
+  const Icon = DecorativeIcons[icon]
+  const iconMargin = (spacingLevel, hasIcon) => {
```
Comment (jraff) [inline packages/CalloutParagraph/CalloutParagraph.jsx:82]: Can we move this function outside of the React component?

File: packages/CalloutParagraph/CalloutParagraph.jsx [hunk 5]
```diff
@@ -92,12 +122,17 @@ CalloutParagraph.propTypes = {
    * Font size and padding around text will be smaller.
    */
   compact: PropTypes.bool,
+  /**
+   * Supply icon name to display before text. Refer to TDS SVGIcon for full list.
+   */
+  icon: PropTypes.string,
```
Comment (jraff) [inline packages/CalloutParagraph/CalloutParagraph.jsx:128]: See https://github.com/telus/tds-core/blob/master/packages/Link/Link.jsx#L159

**Review by jraff — CHANGES_REQUESTED** [review]

**Review by jraff — APPROVED** [review]

**Pull request merged**